cmd/demo.yaml

```diff
@@ -5,187 +5,204 @@ log: debug
+        ret
     maxcurrent:
-      source: js
+      source: go
       vm: shared
       script: |
-        set();
-        var lp = state.loadpoints[0];
-        lp.maxcurrent = val;
-        if (lp.enabled) lp.chargepower = lp.maxcurrent * 230 * lp.phases; else lp.chargepower = 0;
+        lp := &state.Loadpoints[0];
+        lp.MaxCurrent = maxcurrent;
+        if (lp.Enabled) {lp.ChargePower = lp.MaxCurrent * 230 * lp.Phases} else {lp.ChargePower = 0}
 
   - name: charger_2
     type: custom
     enable:
-      source: js
+      source: go
       vm: shared
       script: |
-        set();
-        var lp = state.loadpoints[1];
-        lp.enabled = val;
-        if (lp.enabled) lp.chargepower = lp.maxcurrent * 230 * lp.phases; else lp.chargepower = 0;
+        lp := &state.Loadpoints[1];
+        lp.Enabled = enable;
+        if (lp.Enabled) {lp.ChargePower = lp.MaxCurrent * 230 * lp.Phases} else {lp.ChargePower = 0}
     enabled:
-      source: js
+      source: go
       vm: shared
       script: |
-        state.loadpoints[1].enabled;
+        state.Loadpoints[1].Enabled;
     status:
-      source: js
+      source: go
       vm: shared
       script: |
-        if (state.loadpoints[1].enabled) "C"; else "B";
+        ret := "B"
+        if (state.Loadpoints[1].Enabled) {ret="C"} else {ret="B"}
+        ret
     maxcurrent:
-      source: js
+      source: go
       vm: shared
       script: |
-        set();
-        var lp = state.loadpoints[1];
-        lp.maxcurrent = val;
-        if (lp.enabled) lp.chargepower = lp.maxcurrent * 230 * lp.phases; else lp.chargepower = 0;
+        lp := &state.Loadpoints[1];
+        lp.MaxCurrent = maxcurrent;
+        if (lp.Enabled) {lp.ChargePower = lp.MaxCurrent * 230 * lp.Phases} else {lp.ChargePower = 0}
     phases1p3p:
-      source: js
+      source: go
       vm: shared
       script: |
-        set();
-        if (val === 1) lp.phases = 1; else lp.phases = 3;
-        lp.phases;
+        if (phases == 1) {lp.Phases = 1} else {lp.Phases = 3}
+        lp.Phases;
 
 vehicles:
   - name: vehicle_1
     title: blauer e-Golf
     type: custom
     soc:
-      source: js
+      source: go
       vm: shared
       script: |
-        var lp = state.loadpoints[0];
-        if (lp.chargepower > 0) lp.vehicleSoc+=0.1; else lp.vehicleSoc-=0.1;
-        if (lp.vehicleSoc < 15) lp.vehicleSoc = 80;
-        if (lp.vehicleSoc > 80) lp.vehicleSoc = 15;
-        lp.vehicleSoc;
+        lp := &state.Loadpoints[0];
+        if (lp.ChargePower > 0) {lp.VehicleSoc+=0.1} else {lp.VehicleSoc-=0.1}
+        if (lp.VehicleSoc < 15) {lp.VehicleSoc = 80}
+        if (lp.VehicleSoc > 80) {lp.VehicleSoc = 15}
+        lp.VehicleSoc;
     range:
-      source: js
+      source: go
       vm: shared
       script: |
-        var lp = state.loadpoints[0]
-        var range = (44 * lp.vehicleSoc) / 15;
-        range
+        lp := &state.Loadpoints[0]
+        ret := (44 * lp.VehicleSoc) / 15;
```

Comment:
Das sieht komisch aus

Reply:
Wenn ich das richtig verstanden habe, dann macht `lp := state.Loadpoints[0]` in go ein copy und `lp := &state.Loadpoints[0]`gibt eine Referenz zurück. Bei einigen settern, wird dann in lp.XXX was reingeschrieben, das geht nur mit der referenzsyntax. Ich habe dann überall die referenzsyntax genutzt, damit es einheitlich ist. Sonst kopiert man sich das von einem getter in einen setter und wundert sich, das nichts gesetzt wird. Nötig ist es aber nicht.

Reply:
Dann würde ich die Loadpoints vllt. lieber als `[]*State` deklarieren?

Reply:
wäre aber egal wenn wir die Demoänderung dann nicht committen!